### PR TITLE
Feature-create user signup endpoint-#165984352

### PR DESCRIPTION
**Pull request by Arnoldmn**
#### What does this PR do?
 - This PR creates user sign  up accounts
#### Description of Task to be completed?
POST /api/v1/auth/signup
#### How should this be manually tested?
This PR should be tested using postman or advanced REST
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/165984352
#### Screenshots (if appropriate)

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "presets": [
+        "@babel/preset-env"
+    ]
+}
```
Comment (meshack-mbuvi) [inline .babelrc]: Add a blank line at the end of every file.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+[![Coverage Status](https://coveralls.io/repos/github/Mumbere7/quickCredit/badge.svg?branch=develop)](https://coveralls.io/github/Mumbere7/quickCredit?branch=develop)
+[![Build Status](https://travis-ci.com/Mumbere7/quickCredit.svg?branch=master)](https://travis-ci.com/Mumbere7/quickCredit)
```
Comment (meshack-mbuvi) [inline README.md]: The build status badge should point to `develop` branch. Yours is pointing to `master`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+3. Admin view all current loans (not fully repaid).
+4. Admin view all repaid loans.
+5. Admin view a specific loan application.
+6. Admin approve/reject loan application.
+7. Admin post loan repayment transaction in favour of a client.
+
+> > To view a live demo of this project go to https://mumbere7.github.io/quickCredit/
+> > Give your feed back to the admin in acse of any changes
+
+# <<<<<<< HEAD
```
Comment (meshack-mbuvi) [inline README.md]: It seems you didn't resolve the merge conflicts well. Please resolve them. You can reach out to Murhula Metre for help about this if you get stuck.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+## To have this project just use
+
+1. git clone [project-name]
+2. for a better view i recommend google chrome
+   > > > > > > > 85baa567a588ac1307bfc180dad4634a4355a578
```
Comment (meshack-mbuvi) [inline README.md]: Same case as above(merge conflicts)

File: server/controllers/users.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      password,
+      address,
+      status,
+      isAdmin
+    } = req.body;
+
+    const userExists = users.find(user => user.email === email);
+
+    if (userExists) {
+      res.status(400).json({
```
Comment (meshack-mbuvi) [inline server/controllers/users.js]: 1. The status code here is not correct. Check out for the correct status code when user tries to create an account using an email address which already exists in the system.
2. Line 25 should be `return res.status ...`

File: server/controllers/users.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    const user = {};
+    user.id = users.length + 1;
+    user.email = email;
+    user.firstname = firstname;
+    user.lastname = lastname;
+    user.password = password;
+    user.address = address;
+    user.status = "pending" || "approved";
```
Comment (meshack-mbuvi) [inline server/controllers/users.js]: Line 38 should be `user.status = "pending"`;

File: server/controllers/users.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    jwt.sign(
+      {
+        userId: users.id,
+        email: users.email,
+        password: users.password
+      },
+      "secretkey",
+      (error, token) => {
+        if (error) {
+          return res.status(401).json({
```
Comment (meshack-mbuvi) [inline server/controllers/users.js:53]: The status code here is wrong. Did you read the document I shared on the channel about status codes?

File: server/controllers/users.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+          status: 200,
+          data: {
+            token: user.token,
+            id: user.id,
+            email: user.email,
+            firstname: user.firstname,
+            lastname: user.lastname,
+            password: user.password,
+            address: user.address,
+            status: user.status === "rejected" || "approved",
```
Comment (meshack-mbuvi) [inline server/controllers/users.js]: What are you trying to achieve here?

File: server/controllers/users.js [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+            isAdmin: user.isAdmin === true || false
```
Comment (meshack-mbuvi) [inline server/controllers/users.js]: What will be the values of `isAdmin`?

**Review by meshack-mbuvi — CHANGES_REQUESTED** [review]

File: .babelrc (hunk 1 above)
Comment (Arnoldmn) [reply .babelrc]: Okay thank you, will work on that but with vs code when I leave an empty line at the end of the fil,e it automatically reformats the code.

File: README.md (hunk 2 above)
Comment (Arnoldmn) [reply README.md]: Okay I will do that.

File: README.md (hunk 3 above)
Comment (Arnoldmn) [reply README.md]: I tried but it failed, but I will do as you have requested.

File: server/controllers/users.js (hunk 9 above)
Comment (Arnoldmn) [reply server/controllers/users.js]: When a user enters true it should be true and when they enter false it should false.

File: server/controllers/users.js (hunk 7 above)
Comment (Arnoldmn) [reply server/controllers/users.js:53]: Yes, I read them and understood very well.

File: server/controllers/users.js (hunk 8 above)
Comment (Arnoldmn) [reply server/controllers/users.js]: I am trying to fetch users data after signup.

Comment (meshack-mbuvi) [reply server/controllers/users.js]: What about the `user.status`?

Comment (Arnoldmn) [reply server/controllers/users.js]: It should be unverified instead of rejected or approved

File: server/server.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import express from 'express';
+import { json, urlencoded } from 'body-parser';
+// import swaggerUi from 'swagger-ui-express';
+import userRoute from './routes/users';
+// import loanRoute from './routes/loan';
```
Comment (meshack-mbuvi) [inline server/server.js]: Remove the code that you have commented out please.

File: server/tests/user-signup-test.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+chai.should();
+chai.use(chaiHttp);
+const { expect } = chai;
+
+describe("User sign up", () => {
+  it("Should return a 201 and confirm user a valid input", done => {
+    /**
+     * request new user input
+     */
```
Comment (meshack-mbuvi) [inline server/tests/user-signup-test.js]: No need for these comments. Your code should be self-explanatory.

File: server/tests/user-signup-test.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+    chai
+      .request(app)
+      .post("/api/v1/auth/signup")
+      .send(newUser)
+      .end((error, res) => {
+        expect(res).to.have.status(200);
+        done();
+      });
+  });
+});
```
Comment (meshack-mbuvi) [inline server/tests/user-signup-test.js]: Add a new line at the end of the line, please.

**Review by meshack-mbuvi — CHANGES_REQUESTED** [review]

File: server/server.js (hunk 10 above)
Comment (Arnoldmn) [reply server/server.js]: Okay am going to remove them

File: server/tests/user-signup-test.js (hunk 12 above)
Comment (Arnoldmn) [reply server/tests/user-signup-test.js]: Okay

File: server/tests/user-signup-test.js (hunk 11 above)
Comment (Arnoldmn) [reply server/tests/user-signup-test.js]: Okay

**Pull request merged**